src/interpreter.rs

```diff
@@ -18,6 +64,95 @@ impl Execute for Expression {
     }
 }
 
+#[derive(Debug, Clone, Copy)]
+pub enum OperandItem<'a> {
+    Val(bool),
+    Predicate(&'a Predicate),
+}
+fn evaluate_operand_item(item: OperandItem, ctx: &mut Context, m: &mut Match) -> bool {
+    match item {
+        OperandItem::Val(b) => b,
+        OperandItem::Predicate(p) => p.execute(ctx, m),
+    }
+}
+
+impl Execute for Route {
+    fn execute(&self, ctx: &mut Context, m: &mut Match) -> bool {
+        //operand stack
+        let mut top: usize = 0;
+        let mut operand_stack: [OperandItem; 2] = [OperandItem::Val(false); 2];
+
+        for item in &self.stack {
+            match item {
+                RouteTerm::LogicalOperator(op) => {
+                    match op {
```

Comment:
It's possible to match with `RouteTerm::LogicalOperator(RouteLogicalOperators::And)` Thus we do not need to nest 2 levels of match.